### PR TITLE
Typo/grammar fixes plus copy edits for Rust roadmap (19 files)

### PR DESCRIPTION
**Pull request by blamowizard**
This started as a small effort to fix a few typos I noticed in the Rust roadmap.

However, I ended up spending a few hours tweaking grammar, sentence structure, readability & _markdown_ **emphasis**, and just rewriting stuff to be more informative or concise in areas I thought needed it. Unlikely to work on it any more so submitting the PR.

Anyway, first PR I've submitted in my life, let me know if I should do anything different or if my commit messages suck. Thank you, you're awesome!

### CODE REVIEW

File: src/data/roadmaps/rust/content/101-language-basics/102-constructs/102-traits.md [hunk 1]
```diff
@@ -1,7 +1,11 @@
 # Traits
 
-Traits in Rust programming language define shared behaviors. They are a way to group method signatures together to define a set of behaviors required by a type. Anything with a certain trait will have the behavior of that trait's methods. Traits are abstract, and it's not possible as such to create instances of traits. However, we can define pointers of trait types, and these can hold any implementer of the trait. A trait is implemented for something else, which can be a concrete type or another trait.
+**Traits** in Rust define behaviors that are shared among different data types. Implementing traits for data types is a great way to group method signatures together and define a set of behaviors your types require. Essentially, anything with a certain `trait` applied to it will "inherit" the behavior of that trait's methods, but this is not the same thing as inheritance found in object-oriented programming languages.
```
Comment (dansholds) [inline src/data/roadmaps/rust/content/101-language-basics/102-constructs/102-traits.md:3]: I wouldn't say there's any reason to emphasize "Traits" here as it's I'm the traits topic.

File: src/data/roadmaps/rust/content/101-language-basics/102-constructs/100-enums.md [hunk 2]
```diff
@@ -1,6 +1,10 @@
 # Enums
 
-Enums, short for enumerations, allow you to define a type by enumerating its possible variants. An instance of an enum type could be any one of these variants, and in Rust, enums are incredibly versatile. Unlike enumerations in some other languages, these variants aren't restricted to a singular data type. You can attach data to each variant of the enum, and the data could have different types and amounts for each variant. It's also possible to embed structs and even other enums. Enums in Rust also enable pattern-matching, which allows you to compare a value to a series of patterns, handle it if it matches one of them, and execute code based on which pattern has been matched.
+An **enum** — short for enumeration — is a custom data type that allows you to define a type by enumerating (listing out one-by-one) all of its possible variants. In Rust, if something is one of a given set of possibilities (e.g., `Rock` or `Paper` or `Scissors`), it's probably appropriate to represent that data with an enum, like so: `enum RpsChoice { Rock, Paper, Scissors }`.
```
Comment (dansholds) [inline src/data/roadmaps/rust/content/101-language-basics/102-constructs/100-enums.md:3]: Again, no need to emphasize enum, and let's remove the "-" in favour of a comma such as 

"An enum, short for enumeration, is a custom..."

File: src/data/roadmaps/rust/content/100-introduction/101-why-rust.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # Why use Rust?
 
-Rust is a systems programming language that aims to provide memory safety, concurrency, and performance with a focus on zero cost abstractions. It was originally created by Graydon Hoare at Mozilla Research, with contributions from Brendan Eich, the creator of JavaScript. Rust is appreciated for the solutions it provides to common programming language issues. Its emphasis on safety and speed, the support for concurrent programming, along with a robust type system are just a few reasons why developers choose Rust.
+Rust is a system programming language that aims to provide memory safety, concurrency, and performance with a focus on zero cost abstractions. It was originally created by Graydon Hoare at Mozilla Research, with contributions from Brendan Eich, the creator of JavaScript. Rust is appreciated for the solutions it provides to common programming language issues. Its emphasis on safety, speed, and support for concurrent programming — as well as its robust type system — are just a few reasons why developers choose Rust.
```
Comment (dansholds) [inline src/data/roadmaps/rust/content/100-introduction/101-why-rust.md:3]: Again let's replace the "-" with a comma

File: src/data/roadmaps/rust/content/100-introduction/102-memory-safety.md [hunk 4]
```diff
@@ -1,7 +1,7 @@
...
+Rust: a systems programming language that runs blazingly fast, prevents segfaults, and guarantees thread safety. It is graced with the feature of "memory safety without garbage collection," an attribute that makes Rust one of a kind. "Memory safety" is ensuring that software is not causing any memory leaks or dangling pointers while accessing the system's memory. In Rust, memory safety is accomplished through a system called ownership, with a set of rules that the compiler checks at compile time. This ownership system eliminates the need for garbage collection or manual memory management, thus ensuring swift execution of software and a safer memory environment. Rust's memory management features even support concurrent programming, providing options for shared and mutable state access that ensure thread safety while reducing the risk of thread unsafety.
```
Comment (dansholds) [inline src/data/roadmaps/rust/content/100-introduction/102-memory-safety.md:3]: "Rust is a systems programming..."

**Review by dansholds — CHANGES_REQUESTED** [review]
Hey! Thanks for the efforts that you've put into this!

While I agree with most of the additions, there's some oddities and inconsistencies in grammar used so let's just correct those ☺️

**Review by blamowizard — COMMENTED** [review]
I think clicking “review changes” like this means it adds the commits I made to fix what you suggested to the PR. I’ll find out now 😅

**Pull request merged**